server/postgresql.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/datatypes"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type Postgres struct {
+	db *gorm.DB
+}
+
+type PostgresConfig struct {
+	Host     string
+	Port     int
+	User     string
+	Password string
+	DBName   string
+}
+
+type DBTelemetry struct {
+	ID            int64          `gorm:"column:id; primaryKey"`
+	Application   string         `gorm:"column:application"`
+	TelemetryType string         `gorm:"column:telemetry_type"`
+	Time          time.Time      `gorm:"column:time"`
+	Payload       datatypes.JSON `gorm:"column:payload"`
+}
+
+func NewPostgres(config *PostgresConfig) StorageAdapter {
+	db, err := initDB(config)
+	if err != nil {
+		log.Fatalf("failed to connect database: %s", err)
+		return nil
+	}
+	err = migrateDB(db)
+	if err != nil {
+		log.Fatalf("failed to migrate database: %s", err)
+		return nil
+	}
+	return &Postgres{db: db}
+}
+
+func (p *Postgres) StoreHostTelemetry(h []*HostTelemetry) error {
+	if len(h) == 0 {
+		return nil
+	}
+	var storableTelemetries []*DBTelemetry
+
+	for _, host := range h {
+		telemetryPayload, err := json.Marshal(host)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal telemetry payload")
+		}
+		storableTelemetries = append(
+			storableTelemetries,
+			&DBTelemetry{
+				Application:   "trento",
+				TelemetryType: "host_telemetry",
+				Time:          host.Time,
+				Payload:       telemetryPayload,
+			},
+		)
+	}
+
+	err := p.db.Create(storableTelemetries).Error
+
+	if err != nil {
+		log.Errorf("failed to store host telemetry: %s", err)
+	}
+
+	return err
+}
```

Comment:
Can we name this kind of stuff other than "telemetry"? Telemetry is a thing, our operands are entries, so I'd just name them `entry`, `storableEntry`, or stuff like that.

What do you think?

Reply:
something like
```
func (p *Postgres) StoreHostTelemetry(h []*HostTelemetry) error {
	if len(h) == 0 {
		return nil
	}
	var storableEntries []*DBTelemetry

	for _, entry := range h {
		payload, err := json.Marshal(entry)
		if err != nil {
			return errors.Wrap(err, "failed to marshal telemetry payload")
		}
		storableEntries = append(
			storableEntries,
			&DBTelemetry{
				Application:   "trento",
				TelemetryType: "host_telemetry",
				Time:          entry.Time,
				Payload:       payload,
			},
		)
	}

	err := p.db.Create(storableEntries).Error

	if err != nil {
		log.Errorf("failed to store host telemetry: %s", err)
	}

	return err
}
```

Reply:
@nelsonkopliku exactly! Now could you rename this function `StoreEntry`? IDK, what does @fabriziosestito say about that?

Reply:
Disclaimer: personal and biased opinion after the beep:
_beep_
I'm not a big fan of the `able` terms in the context of variables names, the bias is coming for my background where the `able` was used to name mixins or behaviors in general.
Personally I would have used `telemetries` since it is in the context of the persistence layer and it is easy to infer what we are referring for, also it does not clash with the `h` variable, but it is a matter of style I believe!

Reply:
I don't have anything against the `entry` `storableEntry` change.
@fabriziosestito I feel your pain, anyway this shouldn't cause harms, it that ok with you?

@dottorblaster 
I see what you mean about `StoreEntry`, that could be a way to generalize a bit things, and believe me, I like doing that :sweat_smile: 
It would be really interesting exploring the domain of such a telemetry service and model stuff accordingly, yet I am not sure it is the right time to do this.

Would that be ok for you to defer that abstraction process?

So wrapping up, I propose to only change the `entry` `storableEntry` thingy.

Works?

Reply:
SMASH THE DEFER BUTTON

Reply:
Made the change :rocket:

Reply:
@nelsonkopliku 
thanks! good job btw! we are just bikeshedding our friday with style